### ISSUE TITLE
NcapturingParens incorrectly formatted in AtomEscape definition in Annex B

### ISSUE THREAD

**Issue by michaelficarra** [tc39/ecma262#570]
<img alt="AtomEscape" src="https://cloud.githubusercontent.com/assets/218840/15272018/989e2dde-1a1a-11e6-93ed-45fb478b1cc3.png">

I believe this is due to an ecmarkup/grammarkdown bug.

/cc @disnet

**Comment by littledan** [comment]
cc @rbuckton

**Comment by michaelficarra** [comment]
This has actually gotten worse.

![screenshot from 2018-05-23 14-58-54](https://user-images.githubusercontent.com/218840/40445343-d7cad3fc-5e99-11e8-97a5-1f34f746a63b.png)

Now &le; isn't rendering either.

**Comment by jmdyck** [comment]
It's gotten even worse: now the "but only if" clause doesn't render at all! [link](https://tc39.es/ecma262/#prod-annexB-AtomEscape)

If you go to the [ES2020 release](https://github.com/tc39/ecma262/releases/tag/es2020), it doesn't appear in the version on tc39.es, but does appear in the version at ecma-international.org, and also in the pdf. Not sure if that helps determine when the disappearance occurred.

**Comment by rbuckton** [comment]
I'll take a look.

**Comment by rbuckton** [comment]
I checked into `grammarkdown`, and as far as I can tell I'm emitting an `emu-gmod` for ecmarkup to process:

**Input:**
```grammarkdown
AtomEscape[U, N] ::
    [+U] DecimalEscape
    [~U] DecimalEscape [> but only if the CapturingGroupNumber of |DecimalEscape| is &lt;= _NcapturingParens_]
    CharacterClassEscape[?U]
    CharacterEscape[~U, ?N]
    [+N] `k` GroupName[?U]
```

**Output:**
```ecmarkup
<emu-production name="AtomEscape" params="U, N" type="lexical">
    <emu-rhs a="c727c42b" constraints="+U"><emu-nt>DecimalEscape</emu-nt></emu-rhs>
    <emu-rhs a="f59e7f4b" constraints="~U">
        <emu-nt>DecimalEscape</emu-nt>
        <emu-gmod>but only if the CapturingGroupNumber of <emu-nt>DecimalEscape</emu-nt> is &lt;= _NcapturingParens_</emu-gmod>
    </emu-rhs>
    <emu-rhs a="da3d719e"><emu-nt params="?U">CharacterClassEscape</emu-nt></emu-rhs>
    <emu-rhs a="131b7a46"><emu-nt params="~U, ?N">CharacterEscape</emu-nt></emu-rhs>
    <emu-rhs a="f3e6826b" constraints="+N">
        <emu-t>k</emu-t>
        <emu-nt params="?U">GroupName</emu-nt>
    </emu-rhs>
</emu-production>
```

@bakkot can you take a look at what ecmarkup is doing with this output?

**Comment by rbuckton** [comment]
I've added a test case to grammarkdown for this just to be on the safe side: https://github.com/rbuckton/grammarkdown/commit/fe5722a224e27c3d6de00c726f7dce356ac0efc7

**Comment by ljharb** [comment]
In particular, I'm not sure how https://www.ecma-international.org/ecma-262/11.0/#prod-annexB-AtomEscape and https://tc39.es/ecma262/2020/#prod-annexB-AtomEscape would be different, given that I theoretically generated them at the same time. Indeed, though, the former has `<emu-gmod>but only if the CapturingGroupNumber of <emu-nt id="_ref_16082"><a href="#prod-DecimalEscape">DecimalEscape</a></emu-nt> is &lt;= _NcapturingParens_</emu-gmod>` while the latter lacks it.

**Comment by bakkot** [comment]
Opened https://github.com/tc39/ecmarkup/issues/266. I should have a fix up in a bit.

**Comment by rbuckton** [comment]
Turns out this *is* an issue in `grammarkdown`, as it wasn't treating `[&gt;` as `[>` when parsing a prose assertion. I've published `grammarkdown@3.1.1` which fixes this.
